### DISCUSSION TITLE
linking other tasks is weird

### DISCUSSION THREAD

**Discussion by BenByes** [TechbeeAT/jtxBoard#1503] [General]
I was thrilled to use the app because I read that it can deal with multiple parents for tasks, but it didn't really work for me.

Then I realized that I can't attach subtasks, only top-level.

Example: I have task "save money" and under this task I have subtasks as follow:

- increase salary
- watch expenses

I also have a task "finish semester" that have a subtask as follow:

- make a school project called "the economy today"

Now I created a sub-sub-task for the above task and this sub-sub-task is called "watch a ted-talk about how to save money" which I want to also be a sub-task of "watch expenses" but once I created the ted-talk subtask inside the "school project" task and I want to link it to the "watch expenses" parent it's not possible, because "watch expenses" is a sub-task by itself.

I can, however, link it under the top level task "save money"

is this an expected behavior?

Please consider removing the limitation as it's already possible to link a task to multiple parents as long as they're top level (and that can always be changed later so we get the same result anyway, only it's not intuitive at all)

why is this limitation exists at all?

Please let the user decide how he treats relations, the possibility to link to multiple other items is truly amazing!

pleaseeeeee

and thank you for your unique yet simple treasure "jtx board", the best app with the best potential people don't know about!

all the best

**Comment by patrickunterwegs** [comment]
Hey @BenByes !
To be honest, I haven't added the restriction consciously 😉 I have re-used some existing functionality where I didn't explicitly include subtasks (so I would have had to add the condition, by default I set it to not include them out of the other context).
However, I can easily include them. I just need to maybe add some checks to make sure everything works as expected.

**Reply by BenByes** [reply]
Thanks for honesty lol. I love JTXBoard if, you can make it so I can link parents and subitems not just from the top level but also from different "levels" - that's what I'm asking for and would appreciate it very much so, and I believe it'll be beneficial for more users (there's a similar feature in many apps and it's very successful in these apps, and Jtx is really 95% there and it's even better implemented), I view it as a natural behavior as the app already support multiple connections and multiple subitems levels.

thanks, I'd be happy to test it if required....!

**Reply by patrickunterwegs** [reply]
Hey @BenByes , I have uploaded a new beta-version, it's available on Google Play and here on GitHub in the releases section. It would be great if you could have a look and test a bit =)

**Reply by BenByes** [reply]
Wow that worked so fast, you made my year!

I tested children, sub-children, sub-sub-children, tasks, notes, journals etc

Everything works as expected. thank you!

I'd like to give a brief of how I'm using the app so it may help other folks.

JTX board is a place where I can manage my life without too much hustle. I use the GTD methodology and I have lists of goals, projects, missions etc. I tag tasks with the corresponding categories and resources (type, context, area etc). I use almost every field in the app as long as I can find it later through the filters / grouping e.g categories, resources, collections,, classification, dates etc. I don't use the contacts, alarms, attendees and location fields. I use the attachments fields to link to other apps I use like note apps / google drive where I can store more info about a task / project / notes.

I can then "get wild" with connections between tasks (and also notes / journals) knowing it'll be easy to find them later.

Given the fact that I can now work through much more tasks (and notes) than before (because of the option to attach items from all levels) I have some ideas and consistency suggestions on how to improve the work with the GTD methodology with many tasks and notes without the need for other apps and without wasting some valuable time.

Now where I can write those suggestions? can I write it here and you'll move it to the corresponding thread?

And don't you worry I won't have my expectations too high after the quick fix you just provided for my last request regarding the multiple parents and subitems attaching 😁

**Reply by patrickunterwegs** [reply]
Hey, thanks for your feedback :-)

About other suggestions, if those are small junks and you think they are helpful for everyone, you can directly create an issue. But please make sure you create one issue per specific change. 

If you more like to open discussions on certain topics, please use the discussions here. In this case it's not necessary to strictly create one discussion per idea, but it would be nice if you could group the ideas somehow :-)

**Reply by BenByes** [reply]
done :)

I've posted one discussion divided by three major topics (each topic is rather complicated and contain more than one suggestion but it's kind of a complete ideas so it made since to group those suggestions together)

And then I've posted three issue topics, each contains a rather smaller specific problem, one of them (changing collections for items that have parents or children) is kind of related to the attaching multiple parents feature but it's kind of a bug that was already existed even before.

anyways thanks.. a lot :))

**Reply by patrickunterwegs** [reply]
Thank you, I'll have a look soon :-)
